File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/35_sepcial sort/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	file, _ := os.Open("input.txt")
	defer file.Close()
	oldStd := os.Stdin
	defer func() { os.Stdin = oldStd }()
	os.Stdin = file
	var N int
	fmt.Scanf("%d", &N)

	arr := make([]int, N)
	for i := 0; i < N; i++ {
		fmt.Scanf("%d", &arr[i])
	}

	for i := 0; i < N-1; i++ {
		for j := 0; j < N-1-i; j++ {
			if arr[j] > arr[j+1] {
				if arr[j] < 0 && arr[j+1] < 0 {
					continue
				}
				tmp := arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = tmp
			}
		}
	}
	fmt.Println(arr)
}

File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/35_sepcial sort/main.cpp
#include<stdio.h>

int main() {
	int A[101], N, tmp;
	scanf("%d", &N);

	for (int i = 0; i < N; i++) {
		scanf("%d", &A[i]);
	}

	for (int i = 0; i< N - 1; i++) {
		for (int j=0; j <N-i-1; j++){
			if (A[j] > 0 && A[j +1 ] <0 ){
				tmp = A[j];
				A[j] = A[j+1];
				A[j+1] = tmp;
			}
		}
	}

	for(int i = 0; i<N; i++) printf("%d ",A[i]);

	return 0;
}